cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_train_dense_default_impl.i

```diff
@@ -269,84 +272,102 @@ template <typename algorithmFPType, CpuType cpu>
 void UnorderedRespHelper<algorithmFPType, cpu>::calcImpurity(const IndexType * aIdx, size_t n, ImpurityData & imp) const
 {
     imp.init(_nClasses);
+    algorithmFPType nTotalWeight = 0.;
     for (size_t i = 0; i < n; ++i)
     {
         const ClassIndexType iClass = this->_aResponse[aIdx[i]].val;
-        imp.hist[iClass] += 1;
+        imp.hist[iClass] += this->_aWeight[aIdx[i]].val;
+        nTotalWeight += this->_aWeight[aIdx[i]].val;
     }
-    calcGini(n, imp);
+    calcGini(nTotalWeight, imp);
 }
 
 template <typename algorithmFPType, CpuType cpu>
 void UnorderedRespHelper<algorithmFPType, cpu>::simpleSplit(const algorithmFPType * featureVal, const IndexType * aIdx, TSplitData & split) const
 {
-    split.featureValue = featureVal[0];
     split.left.init(_nClasses);
-    const ClassIndexType iClass1(this->_aResponse[aIdx[0]].val);
-    split.left.hist[iClass1] = 1;
-    split.nLeft              = 1;
-    split.iStart             = 0;
+    const ClassIndexType iClass = this->_aResponse[aIdx[0]].val;
+    split.featureValue          = featureVal[0];
+    split.iStart                = 0;
+    split.left.hist[iClass]     = this->_aWeight[aIdx[0]].val;
+    split.nLeft                 = 1;
+    split.nLeftWeight           = this->_aWeight[aIdx[0]].val;
+    split.nTotalWeight          = this->_aWeight[aIdx[0]].val + this->_aWeight[aIdx[1]].val;
```

Comment:
Double-check SDL conformance

Reply:
it was checked, floating point types of left and right side are similar